frappe/model/create_new.py

```diff
@@ -45,7 +45,9 @@ def make_new_doc(doctype):
 	doc = doc.get_valid_dict(sanitize=False)
 	doc["doctype"] = doctype
 	doc["__islocal"] = 1
-	doc["__unsaved"] = 1
+
+	if not frappe.model.meta.is_single(doctype):
+		doc["__unsaved"] = 1
```

Comment:
should be handled client side? even for Single DocTypes, Customize form is an exception

Reply:
```doc["__unsaved"] = 1``` was added not too long ago here for creation of new documents. Needed, since Frappe doesn't let you save documents until it's set on the client-side.

Single DocTypes are created from the DocType Form; No documents can be created for them. In which case, would you have `__unsaved` be set for a new document created for it?